src/main/js/controls/Graph/Graph.js

```diff
@@ -377,6 +381,82 @@ class Graph extends Construct {
         return this;
     }
 
+    /**
+     * Updates the graph axisData and content.
+     *
+     * @returns {Graph} - Graph instance
+     * @param {Array} graphData - Input array that holds updated values and key
+     */
+    reflow(graphData) {
+        this.config.axis.x.domain = getDomain(
+            this.config.axis.x.type,
+            this.config.axis.x.lowerLimit,
+            this.config.axis.x.upperLimit
+        );
+        const width = getXAxisWidth(this.config);
+        scaleGraph(this.scale, this.config);
+        const axisData = d3.svg
+            .axis()
+            .scale(this.scale.x)
+            .ticks(
+                Math.max(
+                    Math.ceil(width / constants.MAX_TICK_VARIANCE),
+                    constants.MIN_TICKS
+                )
+            )
+            .orient(this.config.axis.x.orientation);
+
+        const svg = d3
```

Comment:
Keep in mind that d3.select or d3.selectAll will return everything in the DOM that matches regardless of which graph it is. For instance, if there are multiple graphs in the page, it would return them all. Instead, restrict it to `this.svg` or something like that which would restrict it to current graph.

Reply:
done here https://github.com/cerner/carbon-graphs/pull/138/commits/9a4183103bd558671c298ca4b33dbc35a2a05f5f.